dashboard/test/test_helper.rb

```diff
@@ -161,6 +161,20 @@ def assert_no_change(expressions, message = nil, &block)
       assert_equal(before[i], e.call, error)
     end
   end
+
+  # Given two hashes, ensure that they have the same set of keys in both
+  # directions, collecting up errors so we can pretty-print them all in one go
+  # if any errors are found.
+  def assert_same_keys(a, b, a_name = 'A', b_name = 'B', message = nil)
```

Comment:
`-` for arrays does what you would want it to, so you could just do something like::
````
assert_equal a.keys, b.keys,  
"#{message}\nkeys in A but not B: #{a.keys - b.keys} \n keys in B but not A: #{b.keys - a.keys}"
````